src/widgets/capture/capturetoolobjects.cpp

```diff
@@ -0,0 +1,131 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// SPDX-FileCopyrightText: 2017-2019 Alejandro Sirgo Rica & Contributors
+
+#include "capturetoolobjects.h"
+
+void CaptureToolObjects::append(const QPointer<CaptureTool>& captureTool)
+{
+    m_captureToolObjects.append(captureTool);
+    m_imageCache.clear();
+}
+
+QPointer<CaptureTool> CaptureToolObjects::at(int index)
+{
+    if (index >= 0 && index < m_captureToolObjects.size()) {
+        return m_captureToolObjects[index];
+    }
+    return nullptr;
+}
+
+void CaptureToolObjects::clear()
+{
+    m_captureToolObjects.clear();
+}
+
+QList<QPointer<CaptureTool>> CaptureToolObjects::captureToolObjects()
+{
+    return m_captureToolObjects;
+}
+
+int CaptureToolObjects::size()
+{
+    return m_captureToolObjects.size();
+}
+
+void CaptureToolObjects::removeAt(int index)
+{
+    if (index >= 0 && index < m_captureToolObjects.size()) {
+        m_captureToolObjects.removeAt(index);
+        m_imageCache.clear();
+    }
+}
+
+int CaptureToolObjects::find(const QPoint& pos, const QSize& captureSize)
+{
+    if (m_captureToolObjects.empty()) {
+        return -1;
+    }
+    QPixmap pixmap(captureSize);
+    pixmap.fill(Qt::transparent);
+    QPainter painter(&pixmap);
+    // first attempt to find at exact position
+    int radius = 3;
```

Comment:
I see radius of 3 and 5 are used throughout this file. Can we define this as a constant in the header?

Reply:
Sure good point, I just forget to refactor this part of code after the debugging.

Reply:
Fixed